android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -1816,28 +1816,36 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
         if (KMManager.shouldAllowSetKeyboard()) {
           if (SystemKeyboard.keyboardPickerEnabled) {
             KeyguardManager km = (KeyguardManager) appContext.getSystemService(Context.KEYGUARD_SERVICE);
+            GlobeKeyAction action = sysKbGlobeKeyAction;
             if(km.inKeyguardRestrictedInputMode()) {
               // Override system keyboard globe key action if screen is locked:
               // 1. Switch to next Keyman keyboard (no menu)
               // 2. When all the Keyman keyboards have been cycled through, advance to the next system keyboard
-              if (sysKbIndexOnLockScreen == -1) {
-                sysKbIndexOnLockScreen = getCurrentKeyboardIndex(context);
-                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD;
-              } else if (sysKbIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
-                sysKbGlobeKeyAction = GlobeKeyAction.GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD;
+              if (sysKbIndexOnLockScreen == getCurrentKeyboardIndex(context)) {
```

Comment:
When would this be `true`? Wouldn't `getCurrentKeyboardIndex()` return the current keyboard that's been changed anyway? Or doesn't `switchToNextKeyboard()` alter that?

Reply:
getCurrentKeyboardIndex() does return the current keyboard (as it's been cycled through `GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD`. 

When it goes all the way around the cycle, (`sysKbIndexOnLockScreen = getCurrentKeyboardIndex()` and the action becomes `GLOBE_KEY_ACTION_ADVANCE_TO_NEXT_SYSTEM_KEYBOARD`

Reply:
Ah snap. It's the `sysKbIndexOnLockScreen` that doesn't change. Perhaps that should be renamed to `sysKbStartingIndexOnLockScreen` or something like that, to clarify?